src/sparse/KokkosSparse_spgemm.hpp

```diff
@@ -92,7 +92,6 @@ void spgemm_numeric(KernelHandle& kh, const AMatrix& A, const bool Amode,
       &kh, A.numRows(), B.numRows(), B.numCols(), A.graph.row_map,
       A.graph.entries, A.values, Amode, B.graph.row_map, B.graph.entries,
       B.values, Bmode, C.graph.row_map, C.graph.entries, C.values);
-  kh.destroy_spgemm_handle();
```

Comment:
Is the handle destroyed when `kh` goes out of scope or was this just a typo?

Reply:
No, kh is passed in (by reference, so it doesn't go out of scope) from the user and it must have had ``create_spgemm_handle(...)`` called on it. Removing this destroy for 2 reasons:
- for symmetry, if the user is responsible for creating the subhandle they should also be responsible for destroying it
- to enable reuse: the simplified interface should be able to do one symbolic followd by multiple numerics

Reply:
The reuse issue affected a user on slack btw, that's how I found out about it